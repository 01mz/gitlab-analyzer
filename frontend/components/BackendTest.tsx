import React, {useState, useEffect} from "react";
import axios, {AxiosResponse} from 'axios';


const BackendTest = () => {
  const [repos, setRepos] = useState([]);

  // Load repo data
  useEffect(() => {
    axios.get(`${process.env.NEXT_PUBLIC_API_URL}/repo`)
      .then((resp: AxiosResponse) => {
        console.log(resp.data);
        setRepos(resp.data);  
      })
  }, [])

  console.log("URL:", process.env.NEXT_PUBLIC_API_URL)
  return (
  <div>
    <h4>API URL: {process.env.NEXT_PUBLIC_API_URL}</h4>
    <p>API data: {JSON.stringify(repos)}</p>
  </div>
)};

export default BackendTest;
