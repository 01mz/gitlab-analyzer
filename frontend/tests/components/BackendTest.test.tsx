import {createShallow} from "@material-ui/core/test-utils";
import React from 'react';
import BackendTest from '../../components/BackendTest';

describe("BackendTest", () =>{
    // @ts-ignore
    let shallow;

    beforeAll(() => {
        shallow = createShallow();
    })

    it("Render BackendTest", () => {
        // @ts-ignore
        const wrapper = shallow(<BackendTest />)
        expect(wrapper).toMatchSnapshot();

    })

})