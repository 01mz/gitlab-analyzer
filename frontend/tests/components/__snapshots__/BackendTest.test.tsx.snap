// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackendTest Render BackendTest 1`] = `
<div>
  <h4>
    API URL: 
  </h4>
  <p>
    API data: 
    []
  </p>
</div>
`;
